### DISCUSSION TITLE
Locale on OpenBSD

### DISCUSSION THREAD

**Discussion by dez11de** [firefly-iii/firefly-iii#6081] [Firefly III]
I have installed Firefly on an OpenBSD machine using httpd as webserver and struggling with the locale.

When I run the command `locale -a` it shows a list of all the supported locales on my machine:
```
mybsd$ locale -a
C
C.UTF-8
POSIX
POSIX.UTF-8
Pig.UTF-8
ar_SD.UTF-8
ar_SY.UTF-8
ca_ES.UTF-8
cs_CZ.UTF-8
da_DK.UTF-8
de_AT.UTF-8
de_CH.UTF-8
de_DE.UTF-8
el_GR.UTF-8
en_AU.UTF-8
en_CA.UTF-8
en_GB.UTF-8
en_US.UTF-8
[snip]
nl_NL.UTF-8
[snip]
```

```
mybsd$ locale -m
UTF-8
```
In the `.env` file I've tried `DEFAULT_LOCALE=` with values `en_US`, `en_US.UTF-8` or `en_US.utf8` but none remove the `Invalid server configuration` message. Preferably I'd like to use Firefly in English language using Dutch locale (`nl_NL.UTF-8`) but if that's not possible I can live with that.
Dates and numbers seem to work mostly fine however, so I'm satisfied when I get rid of the `Invalid server configuration` message.

I'm using Safari on MacOS to acces the server. Language in macOS is set to English (US) and region is set to Netherlands. I don't think Safari has any locale settings.

Any suggestions on getting this setup properly or how to suppress the `Invalid server configuration` message?

I think I read all the documentation about this and the magic word is swordfish.

**Comment by JC5** [comment]
Haha good reading 😁 .

I'm guessing PHP can't access the necessary libraries. I'm not sure why though. Check out the logs. It should have a line that contains "The following array doesnt work", with some options it tried. I'm curious to see what it says.

**Comment by dez11de** [comment]
It says `[2022-05-17 18:32:04] local.ERROR: Could not set locale. The following array doesnt work:  ["en_US.utf8","en_US.UTF-8"]` loads of times... every pageload I assume.

My `php-80.ini` does have this section:
```
[intl]
intl.default_locale = en_US.UTF-8
; This directive allows you to produce PHP errors when some error
; happens within intl functions. The value is the level of the error produced.
; Default is 0, which does not produce any errors.
;intl.error_level = E_WARNING
;intl.use_exceptions = 0
```
The `php-fpm.conf` doesn't contain such a section or anything about locale or intl.

The command `php -m` does show the intl module as being loaded.

Any ideas?

**Reply by JC5** [reply]
Sorry, I don't understand what the problem could be. The locale is literally in the array it's trying to use.

**Reply by dez11de** [reply]
That’s all right. Must be an OpenBSD specific thing.
Is there any way to surpress the message? Everything seems to working fine anyway.

**Reply by JC5** [reply]
There currently isn't, but I'll add it asap!
